Add documentation on recording audio or video from a tab, window or screen in MV3 (#6146)
GoogleChrome/developer.chrome.com: site/en/docs/extensions/reference/tabCapture/index.md

has_warning: This permission <a href="/docs/extensions/mv3/permission_warnings/#permissions_with_warnings">triggers a warning</a>.
---

## Overview

The chrome.tabCapture API allows you to access a [MediaStream][media-stream] containing video and
audio of the current tab. It can only be called after the user invokes an extension, similar to the
behavior of the [activeTab][active-tab] permission.

## Preserving system audio

> When a [MediaStream][media-stream] is obtained for a tab, audio playing in that tab will no longer

Reviewer: ```suggestion
When a [MediaStream][media-stream] is obtained for a tab, audio playback in that tab will no longer
```

be played to the user. This is similar to the behavior of the [get-display-media][getDisplayMedia] API
when the <code>[suppressLocalAudioPlayback][supress-playback]</code> flag is set to true.

To continue playing audio to the user, use the following:

```js
const output = new AudioContext();
const source = output.createMediaStreamSource(stream);
source.connect(output.destination);
```

This creates a new AudioContext and connects the audio of the tab's MediaStream to the default
destination.

## Stream IDs

Calling [chrome.tabCapture.getMediaStreamId][get-media-stream-id] will return a stream ID. To later
access a [MediaStream][media-stream] from the ID, use the following:

```js
navigator.mediaDevices.getUserMedia({
  audio: {
    mandatory: {
      chromeMediaSource: "tab",
      chromeMediaSourceId: id,
    },
  },
  video: {
    mandatory: {
      chromeMediaSource: "tab",
      chromeMediaSourceId: id,
    },
  },
});
```

## Learn More

> To learn more about how to use the chrome.tabCapture API, see

Reviewer: ```suggestion
To learn more about how to use the `chrome.tabCapture` API, see
```

[Audio recording and screen capture][audio-recording-screen-capture]. This demonstrates how to use

> tabCapture and other related APIs to solve a number of common use cases.

Reviewer: ```suggestion
1tabCapture` and related APIs to solve a number of common use cases.
```

has_warning: This permission <a href="/docs/extensions/mv3/permission_warnings/#permissions_with_warnings">triggers a warning</a>.
---

## Overview

The chrome.tabCapture API allows you to access a [MediaStream][media-stream] containing video and
audio of the current tab. It can only be called after the user invokes an extension by clicking or
otherwise triggering the extension's action button, similar to the behavior of the

> [activeTab][active-tab] permission.

Reviewer 2: >invokes an extension by clicking...

I think it goes without saying that clicking an action button invokes an extension

>or otherwise triggering the extension's action button

Is there some other way to trigger a button?

> [activeTab][active-tab] permission.

Author: Yep, it can happen based on a keyboard shortcut or tabbing to the UI and hitting the space bar. But I thought this was the best compromise between hinting at there being different ways and not getting in to the weeds of all of the possibilities, and then "it won't happen if you call `chrome.action.disable`" etc etc...

> [activeTab][active-tab] permission.

Author: I've done the best I can making this clearer. Let me know what you think.

## Preserving system audio

When a [MediaStream][media-stream] is obtained for a tab, audio in that tab will no longer be played

> to the user. This is similar to the behavior of the [get-display-media][getDisplayMedia] API when

Reviewer 2: ```suggestion
to the user. This is similar to the behavior of the [`getDisplayMedia()`][get-display-media] function when
```

the [`suppressLocalAudioPlayback`][supress-playback] flag is set to true.

To continue playing audio to the user, use the following:

```js
const output = new AudioContext();
const source = output.createMediaStreamSource(stream);
source.connect(output.destination);
```

> This creates a new AudioContext and connects the audio of the tab's MediaStream to the default

Reviewer 2: ```suggestion
This creates a new `AudioContext` and connects the audio of the tab's `MediaStream` to the default
```
